libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -35,32 +36,89 @@ pub struct Error {
     pub msg: String,
 }
 
-/// DocPackage represents the documentation for a package and its sub packages
+/// Doc is an enum that can take the form of any form of flux documentation through polymorphism.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct DocPackage {
-    /// the path to the documentation package
-    pub path: String,
-    /// the name of the comments package?
+pub enum Doc {
+    /// Package represents documentation for an entire Flux package.
+    Package(Box<PackageDoc>),
+    /// Value represents documentation for a value exposed from a package.
+    Value(Box<ValueDoc>),
+    /// Builtin represents documentation for a builtin value exposed from a package.
+    Builtin(Box<ValueDoc>),
+    /// Option represents documentation for a option value exposed from a package.
+    Option(Box<ValueDoc>),
+    /// Function represents documentation for a function value exposed from a package.
+    Function(Box<FunctionDoc>),
+}
+
+/// PackageDoc represents the documentation for a package and its sub packages
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PackageDoc {
+    /// the name of the comments package
     pub name: String,
-    /// the doc
-    pub doc: String,
-    /// the values of the comments
-    pub values: Vec<DocValue>,
+    /// the headline of the package
+    pub headline: String,
+    /// the description of the package
+    pub description: Option<String>,
+    /// the members of the package which are values and functions with their parameters
+    pub members: HashMap<String, Doc>,
 }
 
-/// DocValue represents the documentation for a single value within a package.
+/// ValueDoc represents the documentation for a single value within a package.
 /// Values include options, builtins or any variable assignment within the top level scope of a
 /// package.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct DocValue {
-    /// the path to the package
-    pub pkgpath: String,
-    /// the name of the package
+pub struct ValueDoc {
+    /// the name of the value
+    pub name: String,
+    /// the headline of the value
+    pub headline: String,
+    /// the description of the value
+    pub description: Option<String>,
+    /// the type of the value
+    pub flux_type: String,
+}
+
+/// FunctionDoc represents the documentation for a single Function within a package.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FunctionDoc {
+    /// the name of the function
     pub name: String,
-    /// the doc
-    pub doc: String,
-    /// the type of the comments
-    pub typ: String,
+    /// the headline of the function
+    headline: String,
+    /// the description of the function
+    description: String,
+    /// the parameters of the function
+    parameters: Vec<ParameterDoc>,
+    /// the type of the function
+    flux_type: String,
+}
+
+/// Implementation block for FunctionDoc to house future "FunctionDod" methods
+impl FunctionDoc {
+    /// static method to construct a new FunctionDoc
```

Comment:
"Construct a new `FunctionDoc` with name, headline, and flux type." might be better here.